### PR TITLE
Hotfix/fix bugs in editor and performance view

### PR DESCRIPTION
**Pull request by jannessm**

### CODE REVIEW

File: src/app/components/ace-wrapper/ace-wrapper.component.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                        this.song.text.substr(m.index + m[0].length);
+      }
+    });
+  }
+
+  private getNewChord(chord, flat) {
+    let mainKey = chord[0].toLowerCase();
+    let firstTwo = false;
+    if (chord[1] && (chord[1].toLowerCase() === '#' || chord[1].toLowerCase() === 'b')) {
+      mainKey += chord[1].toLowerCase();
```
Comment (nilsroesel) [inline src/app/components/ace-wrapper/ace-wrapper.component.ts]: i would create a variable containing this, also for readability : 
chord[1].toLowerCase()

File: src/app/components/song/song.component.ts [hunk 2]
```diff
@@ -77,4 +77,8 @@ export class SongComponent {
     this.exportService.getPptx(this.song);
   }
 
+  exportPdf() {
+    // this.apiService.generatePdfRequest(this.song.title + '.pdf', this.exportService.getPdf(this.song));
```
Comment (nilsroesel) [inline src/app/components/song/song.component.ts]: ?

File: src/app/dialogs/song-songgroup-form/song-songgroup-form.component.ts [hunk 3]
```diff
@@ -22,11 +26,14 @@ export class SongSonggroupFormComponent implements OnInit {
   id: string;
   songscounter: number[] = [1];
   songs: Song[] = [];
+  filteredSongs: Observable<Song[]>[] = [];
```
Comment (nilsroesel) [inline src/app/dialogs/song-songgroup-form/song-songgroup-form.component.ts]: why observable?

File: src/app/views/editor/editor.component.ts [hunk 4]
```diff
@@ -36,13 +34,21 @@ export class EditorComponent implements OnInit {
...
     if (this.cmdOrCtrlPressed && this.sPressed) {
       this.save();
     }
     this.cmdOrCtrlPressed = false;
     this.sPressed = false;
+
+    const nonUpdateKeys = ['Meta', 'Control', 'Shift', 'Alt'];
+    if (!nonUpdateKeys.find(val => val === key)) {
+      clearTimeout(this.updateInterval);
+      this.updateInterval = setTimeout(() => {
```
Comment (nilsroesel) [inline src/app/views/editor/editor.component.ts]: I dont get this? is this supposed to be a time out or an interval?

Comment (jannessm) [reply src/app/views/editor/editor.component.ts]: it was an interval now its a timeout :D

File: src/app/dialogs/song-songgroup-form/song-songgroup-form.component.ts (hunk 3 above)
Comment (jannessm) [reply src/app/dialogs/song-songgroup-form/song-songgroup-form.component.ts]: because then I can make use of the valuechanges observable from formcontrols

**Pull request merged**